### PR TITLE
(Re)introduce the GLabs Traffic Driver: Carrot

### PR DESCRIPTION
**Pull request by JonNorman**
## What does this change?
This introduces a new ad slot, in which a GLabs traffic driver can be served:

image

This has been AB tested and approved.

In addition to this, I've simplified some of the logic that deals with the optional `absoluteMinAbove` rule. Previously, if this rule was supplied then the measurements of all the candidates were altered, from being relative to the body to being relative to the viewport, when in fact I think we want to always have the measurements relative to the body and then compare against the viewport only if `absoluteMinAbove` is supplied.

## What is the value of this and can you measure success?
Another lever to pull when we want to direct traffic to GLabs campaigns to hit the targets.

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/carrot-traffic-driver.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            minBelow: 100,
+        },
+    },
+    fromBottom: true,
+};
+
+const insertSlot = (paras: Element[]): Promise<void> => {
+    const slot = createSlot('carrot');
+    return fastdom
+        .write(() => paras[0].insertAdjacentElement('beforebegin', slot))
```
Comment (LATaylor-guardian) [inline static/src/javascripts/projects/commercial/modules/carrot-traffic-driver.js]: How do we know paras will not be empty?

**Comment by LATaylor-guardian** [comment]
👍

**Comment by janua** [comment]
:+1:

File: static/src/javascripts/projects/commercial/modules/carrot-traffic-driver.js (hunk 1 above)
Comment (JonNorman) [reply static/src/javascripts/projects/commercial/modules/carrot-traffic-driver.js]: By using the `space-filler` utility it only calls the callback when at least one element has been found, (or more specifically when a `SpaceError` has not been thrown by `spacefinder`).

**Pull request merged**